### PR TITLE
Add backend for DDN's Infinite Memory Engine (IME)

### PR DESCRIPTION
**Pull request by jyvet**
This patch adds the support of IME Native interface
as a new AIORI backend.

### CODE REVIEW

**Review by sydidelot — COMMENTED** [review]
Create

File: src/aiori-IME.c [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        .delete = IME_Delete,
+        .set_version = IME_SetVersion,
+        .fsync = IME_Fsync,
+        .get_file_size = IME_GetFileSize,
+        .access = IME_Access,
+};
+
+/***************************** F U N C T I O N S ******************************/
+
+int IOR_Initialize_IM(int debug_level)
```
Comment (jyvet) [inline src/aiori-IME.c]: I'll remove `IOR_Initialize_IM` and `IOR_Finalize_IM` functions (they are not needed here).

File: src/aiori-IME.c [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+{
+        struct stat stat_buf;
+        IOR_offset_t aggFileSizeFromStat, tmpMin, tmpMax, tmpSum;
+
+        if (ime_native_stat(testFileName, &stat_buf) != 0) {
+                ERR("cannot get status of written file");
+        }
+        aggFileSizeFromStat = stat_buf.st_size;
+
+        if (test->filePerProc == TRUE) {
```
Comment (sydidelot) [inline src/aiori-IME.c]: "== TRUE" can be omitted to improve readability.

File: src/aiori-IME.c [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                fd_oflag |= O_APPEND;
+        if (param->openFlags & IOR_CREAT)
+                fd_oflag |= O_CREAT;
+        if (param->openFlags & IOR_EXCL)
+                fd_oflag |= O_EXCL;
+        if (param->openFlags & IOR_TRUNC)
+                fd_oflag |= O_TRUNC;
+
+        *fd = ime_native_open(testFileName, fd_oflag, 0664);
+        if (*fd < 0) ERR("cannot open file");
```
Comment (jyvet) [inline src/aiori-IME.c]: I'll add a free here.

File: src/aiori-IME.c [hunk 4]
```diff
@@ -0,0 +1,277 @@
+/* -*- mode: c; c-basic-offset: 8; indent-tabs-mode: nil; -*-
+ * vim:expandtab:shiftwidth=8:tabstop=8:
+ */
+/******************************************************************************\
+*                                                                              *
+*        Copyright (c) 2003, The Regents of the University of California       *
```
Comment (sydidelot) [inline src/aiori-IME.c]: Could you please add the DDN copyright below line 6:
Copyright (C) 2018, DataDirect Networks

File: src/aiori-IME.c (hunk 1 above)
Comment (sydidelot) [reply src/aiori-IME.c]: Fine with me :+1:

File: src/aiori-IME.c (hunk 3 above)
Comment (sydidelot) [reply src/aiori-IME.c]: :+1:

File: src/aiori-IME.c (hunk 2 above)
Comment (jyvet) [reply src/aiori-IME.c]: Done! Thanks for the review :-)

File: src/aiori-IME.c (hunk 4 above)
Comment (jyvet) [reply src/aiori-IME.c]: Done

**Review by sydidelot — APPROVED** [review]
Thanks for the patch. Everything looks OK!

**Pull request merged**